### DISCUSSION TITLE
Non-private account system with audit trail

### DISCUSSION THREAD

**Discussion by matthiasdebernardini** [fedimint/fedimint#1460] [Fedimint Modules]
Foreign aid organizations have expressed interest in this idea. It could be used in combination with the ecash module to assure that aid funds actually reach the intended recipient, but once they do the recipient can spend them privately.

**Comment by matthiasdebernardini** [comment]
taken from discord by @justinmoon

**Comment by jkitman** [comment]
This is probably the easiest module to implement that certainly delivers value.  An account can simply be a known pubkey perhaps registered with a name guaranteed to be unique within the fed to make payments easier.

Upon discovering transactions paying this pubkey, the user's client automatically turns them into ecash, from there they can be spent privately.

**Reply by elsirion** [reply]
This kinda already exists in the LN module because I assumed LN gateways wouldn't want to deal with e-cash long term. It's not well-tested though and should not be used for now. We should either add test coverage+replay protection or remove it entirely.

https://github.com/fedimint/fedimint/blob/4da6b39372eddc4742a0509a764d90a25423728c/modules/fedimint-ln/src/contracts/account.rs#L7-L11
